chips/rp2040/src/clocks.rs

```diff
@@ -1038,11 +1038,11 @@ impl Clocks {
         (((source_freq as u64) << 8) / freq as u64) as u32
     }
 
+    #[cfg(all(target_arch = "arm", target_os = "none"))]
     #[inline]
     fn loop_3_cycles(&self, clock: Clock) {
         if self.get_frequency(clock) > 0 {
             let _delay_cyc: u32 = self.get_frequency(Clock::System) / self.get_frequency(clock) + 1;
```

Comment:
One thing to watch out for with these changes is that they can cause `rustc` to conclude functions are unused, depending on the target architecture. I've manually validated this doesn't seem to happen here, and the solution is pretty trivial: just add the offending function calls to the stub implementation as well.

bors r+